tests/bin/sessions.test.ts

```diff
@@ -1,215 +1,322 @@
- * Any side effects need to be undone when the test has completed.
- * Preferably within a `afterEach()` since any cleanup will be skipped inside a failing test.
- *
- * - left over state can cause a test to fail in certain cases.
- * - left over state can cause similar tests to succeed when they should fail.
- * - starting or stopping the agent within tests should be done on a new instance of the polykey agent.
- * - when in doubt test each modified or added test on it's own as well as the whole file.
- * - Looking into adding a way to safely clear each domain's DB information with out breaking modules.
+ * Mock prompts module which is used prompt for password
  */
+jest.mock('prompts');
+const mockedPrompts = mocked(prompts);
 
-describe('Session Token Refreshing', () => {
-  const logger = new Logger('pkStdio Test', LogLevel.WARN, [
+describe('CLI Sessions', () => {
+  const logger = new Logger('status test', LogLevel.WARN, [
     new StreamHandler(),
   ]);
   let dataDir: string;
-  let nodePath: string;
-  let passwordFile: string;
-  let sessionFile: string;
-  let polykeyAgent: PolykeyAgent;
-
-  let tokenBuffer: Buffer;
-  let token: SessionToken;
-  let command: string[];
-  const vaultName = 'TestVault' as VaultName;
-
+  let pkAgentClose;
+  let sessionTokenPath: string;
   beforeAll(async () => {
-    // This handles the expensive setting up of the polykey agent.
+    sessionTokenPath = path.join(global.binAgentDir, config.defaults.tokenBase);
+    pkAgentClose = await testBinUtils.pkAgent();
+  }, global.maxTimeout);
+  afterAll(async () => {
+    await pkAgentClose();
+  });
+  beforeEach(async () => {
     dataDir = await fs.promises.mkdtemp(
       path.join(os.tmpdir(), 'polykey-test-'),
     );
-    nodePath = path.join(dataDir, 'keynode');
-    command = ['vaults', 'list', '-np', nodePath];
-    passwordFile = path.join(dataDir, 'passwordFile');
-    sessionFile = path.join(nodePath, 'token');
-    await fs.promises.writeFile(passwordFile, 'password');
-    polykeyAgent = await PolykeyAgent.createPolykeyAgent({
-      password: 'password',
-      nodePath: nodePath,
-      logger: logger,
+    // Invalidate all active sessions
+    await testBinUtils.pkStdio(
+      ['agent', 'lock'],
+      {
+        PK_NODE_PATH: global.binAgentDir,
+        PK_PASSWORD: global.binAgentPassword,
+      },
+      global.binAgentDir,
+    );
+  });
+  afterEach(async () => {
+    await fs.promises.rm(dataDir, {
+      force: true,
+      recursive: true,
     });
-    await polykeyAgent.vaultManager.createVault(vaultName);
   });
-
-  afterAll(async () => {
-    await polykeyAgent.stop();
-    await polykeyAgent.destroy();
-    await fs.promises.rmdir(dataDir, { recursive: true });
+  test('processes should store session token in session file', async () => {
+    // Run command to set token
+    let exitCode, stdout;
+    ({ exitCode, stdout } = await testBinUtils.pkStdio(
+      ['agent', 'status', '--format', 'json', '--verbose'],
```

Comment:
Won't need `--format json` for this test.

Reply:
I've removed this for all of the tests in this file - I don't believe it's necessary for any of them, but feel free to correct me @emmacasolin

Reply:
Ignore this. Have re-added `--format json` for all tests that have an expectation on the `status` value in stdout. See https://github.com/MatrixAI/js-polykey/pull/296#discussion_r765307682